BEIMA.Backend/ReportService/ReportWriter.cs

```diff
@@ -14,6 +14,76 @@ namespace BEIMA.Backend.ReportService
     /// </summary>
     public static class ReportWriter
     {
+        /// <summary>
+        /// Generates a byte[] containing a file object. The
+        /// file contains a header based on the the props of the deviceType object
+        /// and then the data of all devicesTypes as well as a count of the number of
+        /// devices that are associated with a device type.
+        /// </summary>
+        /// <param name="deviceTypes">List of all device types in the database</param>
+        /// <param name="devices">List of all devices in the database</param>
+        /// <param name="delimiter">String used to seperate every value in the report</param>
+        /// <returns>Byte[] containing a file filled with devicetype data and the number of devices associated with that devicetype</returns>
+        public static byte[] GenerateDeviceTypeReport(List<DeviceType> deviceTypes, List<Device> devices, string delimiter = ",")
+        {
+            if(deviceTypes == null || deviceTypes.Count == 0)
+            {
+                return null;
+            }
+
+            if(devices == null || devices.Count == 0)
```

Comment:
I feel like since we are generating reports specifically for device types, I think it's ok to make device type reports that have no devices. I'm open to discussion on this.

Reply:
I'm good with that as well.